src/config/corsConfig.ts

```diff
@@ -0,0 +1,28 @@
+import express from "express";
+import cors, { CorsOptions } from "cors";
+// Define custom CORS options using interface
+interface CustomCorsOptions {
+    origin?: string | string[];
+    methods?: string | string[];
+    credentials?: boolean;
+  }
+  
+  // Function to configure and apply CORS middleware
+ export  const configureCors = (options?: CustomCorsOptions): express.RequestHandler => {
```

Comment:
({orign= '*', methods = '', credentials = true})

Reply:
Destructure options and set default there to avoid defualt and updated options code